### PR TITLE
reuse identical builds

### PR DESCRIPTION
**Pull request by ddollar**

### CODE REVIEW

**Comment by ddollar** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;98wfA4SQ&#x2F;452-reuse-identical-builds-via-tags"> reuse identical builds via tags</a>

**Comment by awsmsrc** [comment]
makes sense, this will speed builds up a lot
:+1:

File: manifest/build.go [hunk 1]
```diff
@@ -21,7 +21,16 @@ func (m *Manifest) Build(dir, appName string, s Stream, cache bool) error {
 		}
 	}
 
+	buildCache := map[string]string{}
+
 	for _, service := range builds {
+		if bc, ok := buildCache[service.Build.Hash()]; ok {
```
Comment (awsmsrc) [inline manifest/build.go]: I'm not convinced Hash() is necessary 

map[manifest.Build]string{} would probably work

Comment (ddollar) [inline manifest/build.go]: I think you can't use things with a map as a key. See the "Key types" section of https://blog.golang.org/go-maps-in-action

That being said I think this isnt properly hashing the args map.

**Pull request closed without merging**